### PR TITLE
Fix generator usage

### LINKED ISSUE #24: Cannot run the tests in continuous integration
Some tests fail both on stable and on nightly compilers which means we cannot use CI on this crate at the moment, cf #23 and particularly: https://github.com/rust-embedded/nb/runs/801586296?check_suite_focus=true

On nightly we get:
```
---- src/lib.rs -  (line 293) stdout ----
error[E0599]: no method named `resume` found for generator `[generator@src/lib.rs:15:22: 29:6 _]` in the current scope
  --> src/lib.rs:332:16
   |
40 |         blinky.resume();
   |                ^^^^^^ method not found in `[generator@src/lib.rs:15:22: 29:6 _]`

error[E0599]: no method named `resume` found for generator `[generator@src/lib.rs:31:24: 36:6 _]` in the current scope
  --> src/lib.rs:333:18
   |
41 |         loopback.resume();
   |                  ^^^^^^ method not found in `[generator@src/lib.rs:31:24: 36:6 _]`

error: aborting due to 2 previous errors
```

On stable we receive tons of errors about unstable `#[feature]` attributes.

### PR DESCRIPTION
**Pull request by eldruin**
This fixes #24. The generator trait definition changed in the mean time.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -298,7 +298,9 @@
 //! #[macro_use(await)]
 //! extern crate nb;
 //!
-//! use std::ops::Generator;
+//! extern crate core;
```
Comment (therealprof) [inline src/lib.rs:301]:
```suggestion
```

File: src/lib.rs [hunk 2]
```diff
@@ -298,7 +298,9 @@
 //! #[macro_use(await)]
 //! extern crate nb;
```
Comment (therealprof) [inline src/lib.rs:299]: Probably should modernize this, too, while we are at it?

File: src/lib.rs (hunk 1 above)
Comment (eldruin) [reply src/lib.rs:301]: It does not compile otherwise.

File: src/lib.rs (hunk 2 above)
Comment (eldruin) [reply src/lib.rs:299]: With Rust 2018 it does not build due to await being a reserved keyword.

File: src/lib.rs (hunk 1 above)
Comment (therealprof) [reply src/lib.rs:301]: Superweird. Probably needs a thourough overhaul.

**Pull request merged**